### ISSUE TITLE
🐛 BUG: wrangler: Can not upload assets/__node_modules folder to cloudflare pages

### ISSUE THREAD

**Issue by G-Ray** [cloudflare/workers-sdk#3615]
### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

3.2.0 [wrangler]

### What version of Node are you using?

18.x

### What operating system are you using?

Linux

### Describe the Bug

When building a static site with `expo` (with metro bundler) + `expo-router`, some assets are stored at `assets/node_modules`.

`wrangler` seems to define an hardcoded list of paths to ignore here: https://github.com/cloudflare/workers-sdk/blob/113fb7fe86cde0c43a7efa7fc2e58ec8c55d8267/packages/wrangler/src/pages/upload.tsx#L112

I would expect being able to override this list, or configure my own include/exclude list.
See related issue https://github.com/cloudflare/workers-sdk/issues/3176.

This is a blocking issue for me as I can't use wrangler to deploy my website.

**Comment by petebacondarwin** [comment]
Just chiming in on the side here... I see you say with Expo: "some assets are stored at assets/node_modules". If you allowed all node_modules files to be included, you would probably end up deploying a whole lot more than you really wanted to your Pages project.

I think that if this was supported it would be more along the lines of #3176, which is to have include/exclude configuration.
I'm going to close this a basically a duplicate of that issue.

**Comment by Cherry** [comment]
This used to work prior to https://github.com/cloudflare/workers-sdk/pull/2103, so while the feature request in #3176 would indeed solve this, it's still technically a regression from prior functional behaviour.

**Comment by nora-soderlund** [comment]
@petebacondarwin
> I see you say with Expo: "some assets are stored at assets/node_modules". If you allowed all node_modules files to be included, you would probably end up deploying a whole lot more than you really wanted to your Pages project.

That is not true for Expo builds. Web builds in Expo only emits the files that are used, so it essentially extracts a few `.ttf` files from the original node_modules folder from the root folder, and places them in the same directory, but in the `build` directory, meaning no actual modules are transfered to the build folder, only the files that are needed at runtime.

Meaning that I now for example, have to use a forked version of wrangler just to get icons on my web app hosted in Cloudflare Pages, or move to another platform.

---

For anyone reading this with the same issue, [Cloudflare is not open to implementing this](https://github.com/cloudflare/workers-sdk/pull/4710#issuecomment-1881132544), so you can clone my fork and link it locally to your Expo package instead. Or choose a different platform.

https://github.com/nora-soderlund/workers-sdk/tree/main2
https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#building-and-running

**Comment by gartnera** [comment]
Building the wrangler cli was too complicated so I just edited the source like so:

```
[root@AGD2 wrangler-dist]# pwd
/usr/lib/node_modules/wrangler/wrangler-dist
[root@AGD2 wrangler-dist]# sed '/"\*\*\/node_modules",/d' cli.js > cliv2.js
[root@AGD2 wrangler-dist]# diff cli.js cliv2.js
146759d146758
<     "**/node_modules",
[root@AGD2 wrangler-dist]# mv cliv2.js cli.js
```

make sure you only see one change in the diff before committing it.

**Comment by friuns2** [comment]
> Building the wrangler cli was too complicated so I just edited the source like so:
> 
> ```
> [root@AGD2 wrangler-dist]# pwd
> /usr/lib/node_modules/wrangler/wrangler-dist
> [root@AGD2 wrangler-dist]# sed '/"\*\*\/node_modules",/d' cli.js > cliv2.js
> [root@AGD2 wrangler-dist]# diff cli.js cliv2.js
> 146759d146758
> <     "**/node_modules",
> [root@AGD2 wrangler-dist]# mv cliv2.js cli.js
> ```
> 
> make sure you only see one change in the diff before committing it.

thanks

**Comment by vishnuc** [comment]
Same issue , is there a way we can do it in metro.config.js , so assets created with different names ?

**Comment by emily-shen** [comment]
While this won't be added to Pages, you can customise what is ignored at upload in [Workers Assets](https://developers.cloudflare.com/workers/static-assets/) using an `.assetignore` file: https://developers.cloudflare.com/workers/static-assets/binding/#ignoring-assets

**Comment by vishnuc** [comment]
> While this won't be added to Pages, you can customise what is ignored at upload in [Workers Assets](https://developers.cloudflare.com/workers/static-assets/) using an `.assetignore` file: https://developers.cloudflare.com/workers/static-assets/binding/#ignoring-assets

ok with this can i upload via wrangler to pages with file names containing node_modules ? I am able to upload manually all the node_module files but why not wrangler ? wrangler deploy is suppose to save our manual work right ? why not add something to wrangler.toml file so only those who need can do it ?

**Comment by emily-shen** [comment]
@vishnuc To clarify, this is not for Pages, but for [Workers Assets](https://developers.cloudflare.com/workers/static-assets/).

**Comment by thescientist13** [comment]
For anyone else running into this like I did, I had files in my build output that started with _node_modules/_ since I want to support the use case where I want to statically analyze `@import` / url / etc references in CSS files at build time, e.g.
```css
@font-face {
  font-family: "Ubuntu";
  src: url("../../node_modules/@fontsource/ubuntu/files/ubuntu-latin-500-normal.woff2")
    format("truetype");
}
```

And somewhat like [**postcss-import**](https://github.com/postcss/postcss-import), in the case of `url` I would automatically copy that specific file _out_ of _node_modules_ and put it into the build output directly automatically.

That said, a work around I can confirm works was to just rename the start of the file / folder path to _node-modules_

**Comment by vishnuc** [comment]
@thescientist13 this is not viable  , cannot rename everything manually

**Comment by vishnuc** [comment]
instead some one shared a script that renames node_modules automatically during build , try that
